hm_pyhelper/miner_param.py

```diff
@@ -76,25 +65,98 @@ def run_gateway_mfr(args):
         raise ECCMalfunctionException(err_str).with_traceback(e.__traceback__)
 
 
+def get_gateway_mfr_path() -> str:
+    direct_path = os.path.dirname(os.path.abspath(__file__))
+    return os.path.join(direct_path, 'gateway_mfr')
+
+
+def get_gateway_mfr_version() -> Version:
+    """
+    Returns version of gateway_mfr
+    """
+    gateway_mfr_path = get_gateway_mfr_path()
+    command = [gateway_mfr_path, '--version']
+
+    try:
+        run_gateway_mfr_result = subprocess.run(
+            command,
+            capture_output=True,
+            check=True
+        )
+    except Exception as e:
+        err_str = f"Exception occurred on running gateway_mfr --version: {e}"
+        LOGGER.exception(err_str)
+        raise GatewayMFRExecutionException(err_str).with_traceback(e.__traceback__)
+
+    # Parse gateway_mfr version
+    try:
+        version_str = run_gateway_mfr_result.stdout.rpartition(' ')[-1]
+        gateway_mfr_version = Version(version_str)
+
+        return gateway_mfr_version
+    except Exception as e:
+        err_str = f"Exception occurred while parsing gateway_mfr version: {e}"
+        LOGGER.exception(err_str)
+        raise GatewayMFRInvalidVersion(err_str).with_traceback(e.__traceback__)
+
+
+def get_gateway_mfr_command(sub_command: str) -> list:
+    gateway_mfr_path = get_gateway_mfr_path()
+    command = [gateway_mfr_path]
+
+    gateway_mfr_version = get_gateway_mfr_version()
+    if Version('0.1.1') < gateway_mfr_version < Version('0.2.0'):
```

Comment:
@posterzh Not sure if there is any value in backward compatibility considering we ship the gateway-mfr binary along with pyhelper. They being incompatible will be deployment mistake.

Reply:
@posterzh On second that it doesn't hurt, we can keep it for sometime.

Reply:
For feching 0.2.x version, we can allow this PR(https://github.com/NebraLtd/hm-pyhelper/pull/148).
The PRs for bumping the gateway_mfr version are made automatically by the [github action](https://github.com/NebraLtd/hm-pyhelper/blob/9a8891eaaf463f4beb0869f0e9f95e82f52a65a3/.github/workflows/update-gateway-mfr-rs.yml).

[This PR](https://github.com/NebraLtd/hm-pyhelper/pull/174) is just to make the compatibility of `hm-pyhelper` between gateway_mfr versions. And won't hurt.

Reply:
yeah, that has everything for the update. We can merge that after this.